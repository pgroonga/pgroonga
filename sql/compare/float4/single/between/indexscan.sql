CREATE TABLE ids (
  id real
);

INSERT INTO ids VALUES (2.1);
INSERT INTO ids VALUES (7.1);
INSERT INTO ids VALUES (6.1);
INSERT INTO ids VALUES (4.1);
INSERT INTO ids VALUES (5.1);
INSERT INTO ids VALUES (8.1);
INSERT INTO ids VALUES (1.1);
INSERT INTO ids VALUES (10.1);
INSERT INTO ids VALUES (3.1);
INSERT INTO ids VALUES (9.1);

CREATE INDEX grnindex ON ids USING pgroonga (id pgroonga_float4_ops);

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;

EXPLAIN (COSTS OFF)
SELECT id
  FROM ids
 WHERE id BETWEEN (3.0::real) AND (9.0::real)
 ORDER BY id ASC;

SELECT id
  FROM ids
 WHERE id BETWEEN (3.0::real) AND (9.0::real)
 ORDER BY id ASC;

DROP TABLE ids;
