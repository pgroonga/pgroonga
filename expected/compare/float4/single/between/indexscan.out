CREATE TABLE ids (
  id real
);
INSERT INTO ids VALUES (2.1);
INSERT INTO ids VALUES (7.1);
INSERT INTO ids VALUES (6.1);
INSERT INTO ids VALUES (4.1);
INSERT INTO ids VALUES (5.1);
INSERT INTO ids VALUES (8.1);
INSERT INTO ids VALUES (1.1);
INSERT INTO ids VALUES (10.1);
INSERT INTO ids VALUES (3.1);
INSERT INTO ids VALUES (9.1);
CREATE INDEX grnindex ON ids USING pgroonga (id pgroonga_float4_ops);
SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
EXPLAIN (COSTS OFF)
SELECT id
  FROM ids
 WHERE id BETWEEN (3.0::real) AND (9.0::real)
 ORDER BY id ASC;
                       QUERY PLAN                        
---------------------------------------------------------
 Index Scan using grnindex on ids
   Index Cond: ((id >= '3'::real) AND (id <= '9'::real))
(2 rows)

SELECT id
  FROM ids
 WHERE id BETWEEN (3.0::real) AND (9.0::real)
 ORDER BY id ASC;
 id  
-----
 3.1
 4.1
 5.1
 6.1
 7.1
 8.1
(6 rows)

DROP TABLE ids;
